tensorboard/data/server/server.rs

```diff
@@ -288,7 +292,7 @@ fn parse_downsample(downsample: Option<data::Downsample>) -> Result<i64, Status>
             num_points
         )));
     }
-    Ok(num_points)
+    Ok(num_points.try_into().unwrap_or(usize::MAX))
```

Comment:
This only matters if you’re on a 32-bit machine *and* you request at
least 2^32 points *and* your reservoir capacity is at least 2^32 *and*
that many points are actually in the reservoir, right? If any one of
those fails, this is just a lossless conversion (or at least that was
the intent—did I mess it up?).

Do we care about that case? As you say, truncating an RPC with 5 billion
points down to merely 4 billion points feels like the least of your
worries in that scenario, and I wasn’t really even intending to publish
32-bit binaries. TensorFlow doesn’t:
<https://pypi.org/project/tensorflow/#files>

(I would actually believe a claim that it can’t possibly matter, since
protos greater than 2 GiB are unsupported basically everywhere due to
legacy parser bugs.)

Somehow I got the idea that we should prefer 64-bit types to 32-bit
types as a general guideline, though looking through <https://aip.dev>,
I don’t see that advice given. Still, without a good reason to make this
32-bit, I’m weakly inclined to keep it 64-bit (and of course it makes no
difference on the wire, since they’re all varints). WDYT?

Reply:
\*shrug\* I agree it's a marginal edge case that won't occur in practice.  It just seemed like we might as well not even need to spend a line of code doing something to handle the "too large value" edge case when we could avoid it entirely by changing `downsample` and communicate up-front in the RPC API that it can't be unreasonably large.

I guess I'd be a little happier if we are going to keep this line if we change it to just hard-fail a request that sets downsample above a certain limit (either `usize::MAX`, or we could set a lower limit that's closer to the practical range, like 1000 or 10k).  It just feels like doing the truncation to `usxize::MAX` is in principle a weird thing to do since it subverts what the user's requested behavior was in a way that wouldn't be readily obvious to the caller / isn't documented.

Reply:
Sure; it feels awfully marginal to me, indeed. Another reason that this
is impossible to hit: on a 32-bit system, this would just fail to
allocate memory for the response, since even with unbounded swap your
address space simply isn’t large enough. So the potentially confusing
behavior never happens.

But no problem: we can hard-fail the request. Done.